src/System.Management.Automation/engine/Utils.cs

```diff
@@ -833,78 +931,23 @@ private static T GetPolicySettingFromGPOImpl<T>(ConfigScope scope) where T : Pol
                 // The corresponding GPO key exists, then create an instance of T
                 // and populate its properties with the settings
                 object tInstance = Activator.CreateInstance(tType, nonPublic: true);
-                var properties = tType.GetProperties(BindingFlags.Instance | BindingFlags.Public);
                 bool isAnyPropertySet = false;
 
-                string[] valueNames = gpoKey.GetValueNames();
-                string[] subKeyNames = gpoKey.GetSubKeyNames();
-                var valueNameSet = valueNames.Length > 0 ? new HashSet<string>(valueNames, StringComparer.OrdinalIgnoreCase) : null;
-                var subKeyNameSet = subKeyNames.Length > 0 ? new HashSet<string>(subKeyNames, StringComparer.OrdinalIgnoreCase) : null;
-
-                foreach (var property in properties)
+                // if PolicySettingFallbackKey is Not set - use PowerShell Core policy reg key
+                if ((int)gpoKey.GetValue(PolicySettingFallbackKey, 0) == 0)
                 {
-                    string settingName = property.Name;
-                    object rawRegistryValue = null;
-
-                    // Get the raw value from registry.
-                    if (valueNameSet != null && valueNameSet.Contains(settingName))
-                    {
-                        rawRegistryValue = gpoKey.GetValue(settingName);
-                    }
-                    else if (subKeyNameSet != null && subKeyNameSet.Contains(settingName))
-                    {
-                        using (RegistryKey subKey = gpoKey.OpenSubKey(settingName))
-                        {
-                            if (subKey != null) { rawRegistryValue = subKey.GetValueNames(); }
-                        }
-                    }
-
-                    // Get the actual property value based on the property type.
-                    // If the final property value is not null, then set the property.
-                    if (rawRegistryValue != null)
+                    isAnyPropertySet = TrySetPolicySettingsFromRegistryKey(tInstance, tType, gpoKey);
+                }
+                else
+                {
+                    // when PolicySettingFallbackKey flag is set (REG_DWORD "1") use Windows PS policy reg key
+                    WindowsPowershellGroupPolicyKeys.TryGetValue(tType.Name, out string winPowershellGpoKeyPath);
+                    Diagnostics.Assert(winPowershellGpoKeyPath != null, StringUtil.Format("The Windows PS GPO registry key path should be pre-defined for {0}", tType.Name));
```

Comment:
This should probably be a resource

Reply:
I think it makes no sense. It is used only in debug code.

Reply:
Yup, I think I misread the code the first time.